domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/FeatureFlagsLoggerTest.kt

```diff
@@ -0,0 +1,198 @@
+import org.oppia.android.testing.platformparameter.TEST_FEATURE_FLAG
+import org.oppia.android.testing.platformparameter.TEST_FEATURE_FLAG_WITH_ENABLED_DEFAULTS
+import org.oppia.android.testing.platformparameter.TestPlatformParameterModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestCoroutineDispatchers
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.caching.AssetModule
+import org.oppia.android.util.data.DataProvidersInjector
+import org.oppia.android.util.data.DataProvidersInjectorProvider
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EnableConsoleLog
+import org.oppia.android.util.logging.EnableFileLog
+import org.oppia.android.util.logging.GlobalLogLevel
+import org.oppia.android.util.logging.LogLevel
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [FeatureFlagsLogger]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = FeatureFlagsLoggerTest.TestApplication::class)
+class FeatureFlagsLoggerTest {
+  @Inject lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+  @Inject lateinit var featureFlagsLogger: FeatureFlagsLogger
+  @Inject lateinit var fakeAnalyticsEventLogger: FakeAnalyticsEventLogger
+
+  @field:[Inject EnableTestFeatureFlag]
+  lateinit var testFeatureFlag: PlatformParameterValue<Boolean>
+  @field:[Inject EnableTestFeatureFlagWithEnabledDefault]
+  lateinit var testFeatureFlagWithEnabledDefault: PlatformParameterValue<Boolean>
+
+  @Before
+  fun setup() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testLogFeatureFlags_logFeatureFlags_hasCorrectSessionId() {
+    featureFlagsLogger.logAllFeatureFlags(TEST_SESSION_ID)
+    testCoroutineDispatchers.runCurrent()
+
+    val eventLog = fakeAnalyticsEventLogger.getMostRecentEvent()
+    assertThat(eventLog).hasFeatureFlagContextThat {
+      hasSessionIdThat().isEqualTo(TEST_SESSION_ID)
+    }
+  }
+
+  @Test
+  fun testLogFeatureFlags_logsTestFeatureFlag_hasCorrectDefaultValues() {
+    featureFlagsLogger.setFeatureFlagItemMap(
+      mapOf(TEST_FEATURE_FLAG to testFeatureFlag)
+    )
+    featureFlagsLogger.logAllFeatureFlags(TEST_SESSION_ID)
+
+    testCoroutineDispatchers.runCurrent()
+
+    val eventLog = fakeAnalyticsEventLogger.getMostRecentEvent()
+    assertThat(eventLog).hasFeatureFlagContextThat {
+      hasFeatureFlagItemContextThat {
+        hasFeatureFlagItemContextThat().isNotNull()
```

Comment:
This is not necessary since the other assertions would not pass if this were null so it is implied.
```suggestion
     
```